include/zephyr/gnss/rtk/decoder.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2025 Croxel Inc.
+ * Copyright (c) 2025 CogniPilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_GNSS_RTK_DECODER_H_
+#define ZEPHYR_INCLUDE_GNSS_RTK_DECODER_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+/**
+ * @brief Get an RTK frame from buffer
+ *
+ * Used by RTK clients to extract frames from a data-buffer.
+ *
+ * @param[in] buf Buffer holding encoded data.
+ * @param[in] buf_len Buffer length.
+ * @param[out] data Pointer to the decoded frame.
+ * @param[out] data_len Length of the decoded frame
+ *
+ * @return Zero if successful.
+ * @return -ENOENT if no frames have been decoded successfully.
+ * @return -EAGAIN if there's an incomplete frame starting at the data pointer.
```

Comment:
you haven't implemented that error code tho?

Reply:
Removed